lib/http/errors.rb

```diff
@@ -19,6 +19,9 @@ class StateError < ResponseError; end
   # Generic Timeout error
   class TimeoutError < Error; end
 
+  # Timeout when first establishing the conncetion
+  class ConnectionTimeoutError < TimeoutError; end
```

Comment:
Wondering if `OpenTimeoutError` might read better as it alines with what `Net::HTTP` has in place. Collective memory and all.

Reply:
Neat. I like this. Makes it more intuitive for folks with the underlying syscall on mind.